### PR TITLE
Arc 2112 catch 405 on other jira api 2

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
1. Try/catch 405 for other similar jira api as well.
2. Merge into the previous logic to convert it into 404 instead of silently mark it as success
3. Modify the backfill to accept 404 as a success

**Why**
That try/catch 405 works well on the devinfo, would like to apply it to other jira api.

**Added feature flags**
N/A

**Affected issues**  
N/A

**How has this been tested?**  
Unit test.

**Whats Next?**

### CODE REVIEW

File: src/jira/client/axios.ts [hunk 1]
```diff
@@ -173,20 +162,24 @@ const instrumentFailedRequest = (baseURL: string, logger: Logger) => {
 		}
 		instrumentRequest(baseURL)(error?.response);
 
-		if (error.response?.status === 503 || error.response?.status === 405) {
+		if (error?.request?.method === "GET" && error?.response?.status === 405) {
+			/**
+			 * When sending a POST request to `https://OLD.atlassian.net/rest/deployments/0.1/bulk` or similar api on that jirahost,
+			 * it sends a 302 response, and redirects to a GET request `https://NEW.atlassian.net/rest/deployments/0.1/bulk`
+			 * This GET request is now failing with a 405 response, which is not an error on the app side
+			 * So treating this as a non-error case
```
Comment (bgvozdev) [inline src/jira/client/axios.ts:170]: I was wondering if it makes sense to reconfigure axios not to follow redirects and to POST using the new hostname?...

Comment (gxueatlassian) [reply src/jira/client/axios.ts:170]: Yup, thought about that, make sense, but in this case, the jira host is redirected, not much point of still sending data to old jirahost.

Comment (gxueatlassian) [reply src/jira/client/axios.ts:170]: And if we follow the redirect and post, we will ends up sending twice the copy of data into new jira host, not sure what will that mean.

**Comment by gxueatlassian** [comment]
Will raise another one.

**Pull request closed without merging**